### PR TITLE
Create autospec guide and autospec-about to fulfill JIRAs:

### PR DESCRIPTION
**Pull request by mvincerx**
- CLEAR 3076: Autospec concept draft
- CLEAR 3077: Autospec guide

Signed-off-by: Michael Vincerra <michaelx.vincerra@intel.com>

### CODE REVIEW

**Comment by mvincerx** [comment]
Please review first draft of autospec guide and autospec-about.  

Notes: 
- In autospec.rst, after "Create a RPM with autospec," an excerpt of `autospec overview` will appear (but will not render in GitHub). 
- I will take the AR of writing a separate guide on setting up a web server (with web-server-basic). Presently, this is missing .

File: source/clear-linux/concepts/autospec-about.rst [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+#. ``autospec`` detects any missed definitions in the ``spec`` file. 
+
+.. note:: 
+
+   * If there are missed definitions, ``autospec`` creates another ``mock``
+     ``chroot`` and starts building again at Step 1. 
+   * If a build error occurs, ``autospec`` stops for user inspection. 
+   * If no build errors occur, RPM packages are successfully built.       
+
+``autospec`` continuously runs updated builds, based on new information
```
Comment (phmccarty) [inline source/clear-linux/concepts/autospec-about.rst]: Describing autospec's build rounds as "updated builds" seems a bit unclear to me. Maybe rephrase as "autospec continues to rebuild the package...".

File: source/clear-linux/concepts/autospec-about.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
+.. _autospec-about: 
+
+.. _autospec-overview: 
+
+Autospec overview
+#################
+
+Whereas a standard RPM build process using ``rpmbuild`` requires a tarball 
+and ``spec`` file to start, ``autospec`` only requires a tarball.
```
Comment (phmccarty) [inline source/clear-linux/concepts/autospec-about.rst]: Autospec also requires a package name in addition to the tarball URI.

File: source/clear-linux/concepts/autospec-about.rst [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+Autospec attempts to infer the requirements of the ``spec`` file. If 
+autospec infers correctly, the control files (Table 1) will automatically 
+correct the build requirements. These control files are used to influence
+the ``spec`` file generation. 
+
+#. The :command:`make autospec` command generates a ``spec`` file from the 
+   control files.  
+
+#. ``autospec`` creates a ``chroot`` with ``mock`` config.
```
Comment (phmccarty) [inline source/clear-linux/concepts/autospec-about.rst]: I prefer the term "build root" instead of "chroot", so as not to be confused with the chroot command.

File: source/clear-linux/concepts/autospec-about.rst [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+the ``spec`` file generation. 
+
+#. The :command:`make autospec` command generates a ``spec`` file from the 
+   control files.  
+
+#. ``autospec`` creates a ``chroot`` with ``mock`` config. 
+   
+#. ``autospec`` attempts to build an RPM from the generated ``spec`` file.
+   
+#. ``autospec`` detects any missed definitions in the ``spec`` file.
```
Comment (phmccarty) [inline source/clear-linux/concepts/autospec-about.rst]: Hmm, maybe "declarations" instead of "definitions"? A spec file includes a variety of information (macro definitions, key/value pairs, file lists, embedded scripts, etc), but much of the non-code pieces are more declarative in nature.

File: source/clear-linux/concepts/autospec-about.rst [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+#. ``autospec`` creates a ``chroot`` with ``mock`` config. 
+   
+#. ``autospec`` attempts to build an RPM from the generated ``spec`` file.
+   
+#. ``autospec`` detects any missed definitions in the ``spec`` file. 
+
+.. note:: 
+
+   * If there are missed definitions, ``autospec`` creates another ``mock``
```
Comment (phmccarty) [inline source/clear-linux/concepts/autospec-about.rst]: Same comment here about "declarations" vs "definitions".

File: source/clear-linux/guides/maintenance/autospec.rst [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+
+   .. code-block:: bash
+
+      cd ~/clearlinux
+
+#. Enter the command: 
+
+   .. code-block:: bash
+
+      make autospecnew URL="https://github.com/bktan/helloclear/archive/helloclear-4.tar.gz" NAME="helloclear"
```
Comment (phmccarty) [inline source/clear-linux/guides/maintenance/autospec.rst]: URL should be `https://github.com/bktan8/helloclear/archive/helloclear-4.tar.gz`

**Comment by mvincerx** [comment]
@bktan8 Will you please let me know when you've created the new helloclear repo? 
https://jira01.devtools.intel.com/browse/OTCIS-15725 is now closed. Thanks.

**Comment by mvincerx** [comment]
@bktan8. (Previous comment was deleted). Please just ping me when you add helloclear repo to clearlinux.

**Comment by bktan8** [comment]
@mvincerx - here you go: 

make autospecnew URL="https://github.com/clearlinux/helloclear/archive/helloclear-v1.0.tar.gz" NAME="helloclear"

**Comment by mvincerx** [comment]
@bktan8 provided new URL for helloclear; I updated in last push. (No other comments in 14 days.)
I will merge this PR by 13:00 on Thurs. 7/12. If you have any feedback, please provide it before that time.
